sale_stock_medical_prescription/models/medical_prescription_order_line.py

```diff
@@ -0,0 +1,146 @@
+        compute='_compute_dispensings',
+        help='Amount pending dispense (using medicine dosage)',
+    )
+    exception_dispense_qty = fields.Float(
+        store=True,
+        compute='_compute_dispensings',
+        help='Qty of dispense exceptions (using medicine dosage)',
+    )
+    cancelled_dispense_qty = fields.Float(
+        store=True,
+        compute='_compute_dispensings',
+        help='Dispense qty cancelled (using medicine dosage)',
+    )
+    active_dispense_qty = fields.Float(
+        store=True,
+        compute='_compute_can_dispense_and_qty',
+        help='Total amount of dispenses that are active in some way',
+    )
+    can_dispense = fields.Boolean(
+        store=True,
+        compute='_compute_can_dispense_and_qty',
+        help='Can this prescription be dispensed?',
+    )
+    can_dispense_qty = fields.Float(
+        store=True,
+        compute='_compute_can_dispense_and_qty',
+        help='Amount that can be dispensed (using medicine dosage)',
+    )
+
+    @api.model
+    def _default_dispense_uom_id(self):
+        return self.env['product.uom'].browse(
+            self.env['product.template']._get_uom_id()
+        )
+
+    @api.multi
+    @api.depends(
+        'dispense_uom_id',
+        'sale_order_line_ids',
+        'sale_order_line_ids.procurement_ids.product_uom',
+        'sale_order_line_ids.procurement_ids.product_qty',
+        'sale_order_line_ids.procurement_ids.state',
+    )
+    def _compute_dispensings(self):
+        for record in self:
+
+            dispense_ids = []
+            dispense_qty = 0.0
+            pending_qty = 0.0
+            cancel_qty = 0.0
+            except_qty = 0.0
+            last_procurement_id = None
+
+            order_line_ids = record.sale_order_line_ids.sorted(
+                key=lambda r: r.order_id.date_order
+            )
+            for line_id in order_line_ids:
+                procurement_ids = line_id.procurement_ids.sorted(
+                    key=lambda r: r.date_planned
+                )
+                for proc_id in procurement_ids:
+
+                    dispense_ids.append(proc_id.id)
+                    last_procurement_id = proc_id
+
+                    if proc_id.product_uom.id != record.dispense_uom_id.id:
+                        _qty = self.env['product.uom']._compute_qty_obj(
+                            proc_id.product_uom,
+                            proc_id.product_qty,
+                            record.dispense_uom_id,
+                        )
+                    else:
+                        _qty = proc_id.product_qty
+
+                    if proc_id.state == 'done':
+                        dispense_qty += _qty
+                    elif proc_id.state in ['confirmed', 'running']:
+                        pending_qty += _qty
+                    elif proc_id.state == 'cancel':
+                        cancel_qty += _qty
+                    else:
+                        except_qty += _qty
+
```

Comment:
From here to L91 - I'm not a big fan of using `_id` in variable names, since it's an Instance of that type, not just the ID.